### ISSUE TITLE
Improve checks for PEP 604 - alternative Union syntax

### ISSUE THREAD

**Issue by cdce8p** [PyCQA/pylint#4065]
### Is your feature request related to a problem?
Pylint should be able to detect if PEP 604 (alternative Union Syntax) is used inappropriately.
The PEP will be implemented in **3.10**, so the following should emit error messages in versions **3 - 3.9**:
```py
Alias = str | int  # Should be an error

var: str | int  # Should be an error

def func(arg: int | str):  # Should be an error
    pass

def func2() -> int | str:  # Should be an error
    pass
```

Each line is a `TypeError` in Python, eg:
```
Traceback (most recent call last):
  File "test.py", line 3, in <module>
    Alias = str | int
TypeError: unsupported operand type(s) for |: 'type' and 'type'
```

**With postponed evaluation of annotations enabled (3.7 - 3.9)**
```py
from __future__ import annotations

Alias = str | int  # Should be an error

var: str | int

def func(arg: int | str):
    pass

def func2() -> int | str:
    pass
```
Only the alias definition is evaluated directly and thus an error.

### Current situation
No errors are emitted, even in versions prior to 3.10 and without the future import.

--
https://www.python.org/dev/peps/pep-0604/
